packages/esm-ward-app/src/config-schema.ts

```diff
@@ -1,5 +1,120 @@
-import { type ConfigSchema } from '@openmrs/esm-framework';
+import { type PersonAddress, Type, validator, validators, type ConfigSchema } from '@openmrs/esm-framework';
+import { type BentoElementType, bentoElementTypes } from './types';
 
-export const configSchema: ConfigSchema = {};
+const defaultWardPatientCard: WardPatientCardDefinition = {
+  card: {
+    id: 'default-card',
+    header: ['bed-number', 'patient-name', 'patient-age', 'patient-address'],
+  },
+  appliedTo: null,
+};
 
-export interface ConfigObject {}
+export const defaultBentoElementConfig: BentoElementConfig = {
+  addressFields: ['cityVillage', 'country'],
+};
+
+export const builtInBentoElements: BentoElementType[] = [
+  'bed-number',
+  'patient-name',
+  'patient-age',
+  'patient-address',
+  'admission-time',
+];
+
+export const configSchema: ConfigSchema = {
+  wardPatientCards: {
+    _description: 'Configure the display of ward patient cards',
+    bentoElementDefinitions: {
+      _type: Type.Array,
+      _default: [],
+      _elements: {
+        id: {
+          _type: Type.String,
+          _description: 'The unique identifier for this custom bento element',
+        },
+        elementType: {
+          _type: Type.String,
+          _description: 'The bento element type',
+          _validators: [validators.oneOf(bentoElementTypes)],
+        },
+        config: {
+          addressFields: {
+            _type: Type.Array,
+            _description: 'For bentoElementType "patient-address", defining which address fields to show',
+          },
+        },
+      },
+    },
+    cardDefinitions: {
+      _type: Type.Array,
+      _default: [defaultWardPatientCard],
+      _description: `An array of card configuration. A card configuration can be applied to different ward locations.
+         If multiple card configurations apply to a location, only the first one is chosen.`,
+      _elements: {
+        id: {
+          _type: Type.String,
+          _description: 'The unique identifier for this card definition. Currently unused, but that might change.',
+        },
+        card: {
+          header: {
+            _type: Type.Array,
+            _element: {
+              _type: Type.String,
+              _description: 'The ID of the (bulit-in or custom) bento element',
+              _validators: [
+                validator(
+                  (bentoElementId: string) => {
+                    const validBentoElementIds: string[] = [...bentoElementTypes];
+                    return validBentoElementIds.includes(bentoElementId);
+                  },
+                  (bentoElementId: string) => 'Invalid bento element id: ' + bentoElementId,
+                ),
+              ],
```

Comment:
```suggestion
              _validators: [validators.oneOf(bentoElementTypes)],
```

(I think that syntax is right, sorry if it's not quite)

If this gives you a strange type error try upgrading core; I fixed the typing of that validator in the last week or two.